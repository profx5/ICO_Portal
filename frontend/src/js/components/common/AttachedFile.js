import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {media} from 'js/utils/media';


const AttachedFile = ({fileName, fileSize, sizeUnits, id, removable, removeHandler, style}) => {
    return (
        <Wrapper className="attached-file" id={id} style={style}>
            <span className="file-name">{fileName}</span>
            {fileSize && <span className="file-size"> ({fileSize} {sizeUnits})</span>}
            {removable && 
                <div onClick={removeHandler} className="file-close">
                    <svg xmlns="http://www.w3.org/2000/svg" width="10" height="10" viewBox="0 0 10 10">
                        <g fill="#C8C8C8" fillRule="evenodd">
                            <path d="M.05 1.464L1.464.05 9.95 8.536 8.536 9.95z"/>
                            <path d="M1.464 9.95L.05 8.536 8.536.05 9.95 1.464z"/>
                        </g>
                    </svg>
                </div>
            }
        </Wrapper>
    )
}


AttachedFile.propTypes = {
    fileName: PropTypes.string,
    fileSize: PropTypes.number,
    sizeUnits: PropTypes.string,
    id: PropTypes.number,
    removable: PropTypes.bool,
    removeHandler: PropTypes.func.isRequired,
    style: PropTypes.object,
}

export default AttachedFile;

const Wrapper = styled.div`
    width: 285px;
    background: rgb(245, 245, 245);
    padding: 0 35px 0 13px;
    height: 36px;
    margin-bottom: 4px;
    float: left;
    clear: left;
    position: relative;
    display: flex;
    justify-content: flex-start;
    ${media.xs} {
        width: 100%;
        min-width: unset;
    }
    .file-name {
        font-size: 16px;
        color: rgb(92, 141, 245);
        letter-spacing: 0.5px;
        font-weight: 600;
        line-height: 36px;
        display: block;
        max-width: 150px;
        overflow: hidden;
        text-overflow: ellipsis;
        ${media.xs} {
            font-size: 14px;
        }
    }
    .file-size {
        font-size: 16px;
        line-height: 36px;
        width: 83px;
        margin-left: 6px;
        white-space: nowrap;
    }
    .file-close {
        position: absolute;
        right: 13px;
        top: 50%;
        transform: translateY(-50%);
        cursor: pointer;
    }
`;
