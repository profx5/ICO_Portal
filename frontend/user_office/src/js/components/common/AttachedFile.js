import React from 'react'
import styled from 'styled-components';
import {media} from './../../utils/media';

import Utils from './../../utils/index';


const AttachedFile = ({fileName, fileSize, id, removable, onRemoveHandler, style}) => {
    return (
        <Wrapper className="attached-file" id={id} style={style}>
            <span className="file-name">{fileName}</span>
            {fileSize && <span className="file-size"> ({Utils.formatFileSize(fileSize).size} {Utils.formatFileSize(fileSize).units})</span>}
            {removable && 
                <div onClick={onRemoveHandler} className="file-close">
                    <svg xmlns="http://www.w3.org/2000/svg" width="10" height="10" viewBox="0 0 10 10">
                        <g fill="#C8C8C8" fillRule="evenodd">
                            <path d="M.05 1.464L1.464.05 9.95 8.536 8.536 9.95z"/>
                            <path d="M1.464 9.95L.05 8.536 8.536.05 9.95 1.464z"/>
                        </g>
                    </svg>
                </div>
            }
        </Wrapper>
    )
}


export default AttachedFile;

const Wrapper = styled.div`
    min-width: 240px;
    background: rgb(245, 245, 245);
    padding: 0 35px 0 13px;
    height: 36px;
    margin-bottom: 4px;
    float: left;
    clear: left;
    position: relative;
    ${media.xs} {
        width: 100%;
    }
    .file-name {
        font-size: 16px;
        color: rgb(92, 141, 245);
        letter-spacing: 0.5px;
        font-weight: 600;
        line-height: 36px;
        ${media.xs} {
            font-size: 14px;
        }
    }
    .file-size {
        font-size: 16px;
        line-height: 36px;
    }
    .file-close {
        position: absolute;
        right: 13px;
        top: 50%;
        transform: translateY(-50%);
        cursor: pointer;
    }
`;
