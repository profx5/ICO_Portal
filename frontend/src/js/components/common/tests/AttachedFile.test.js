import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import AttachedFile from 'js/components/common/AttachedFile';


configure({ adapter: new Adapter() });

it('should render stateless AttachedFile', () => {

    const props = {
        removeHandler: jest.fn()
    }

    const component = shallow(<AttachedFile {...props}/>);
    expect(component).toMatchSnapshot();
})
