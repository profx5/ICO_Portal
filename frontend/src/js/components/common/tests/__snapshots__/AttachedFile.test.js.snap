// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless AttachedFile 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AttachedFile
    removeHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span
          className="file-name"
        />,
        undefined,
        undefined,
      ],
      "className": "attached-file",
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "file-name",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      undefined,
      undefined,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    width: 285px;
    background: rgb(245, 245, 245);
    padding: 0 35px 0 13px;
    height: 36px;
    margin-bottom: 4px;
    float: left;
    clear: left;
    position: relative;
    display: flex;
    justify-content: flex-start;
    ",
          "@media (max-width: 767px)",
          " {
        width: 100%;
        min-width: unset;
    }
    .file-name {
        font-size: 16px;
        color: rgb(92, 141, 245);
        letter-spacing: 0.5px;
        font-weight: 600;
        line-height: 36px;
        display: block;
        max-width: 150px;
        overflow: hidden;
        text-overflow: ellipsis;
        ",
          "@media (max-width: 767px)",
          " {
            font-size: 14px;
        }
    }
    .file-size {
        font-size: 16px;
        line-height: 36px;
        width: 83px;
        margin-left: 6px;
        white-space: nowrap;
    }
    .file-close {
        position: absolute;
        right: 13px;
        top: 50%;
        transform: translateY(-50%);
        cursor: pointer;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span
            className="file-name"
          />,
          undefined,
          undefined,
        ],
        "className": "attached-file",
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "file-name",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        undefined,
        undefined,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    width: 285px;
    background: rgb(245, 245, 245);
    padding: 0 35px 0 13px;
    height: 36px;
    margin-bottom: 4px;
    float: left;
    clear: left;
    position: relative;
    display: flex;
    justify-content: flex-start;
    ",
            "@media (max-width: 767px)",
            " {
        width: 100%;
        min-width: unset;
    }
    .file-name {
        font-size: 16px;
        color: rgb(92, 141, 245);
        letter-spacing: 0.5px;
        font-weight: 600;
        line-height: 36px;
        display: block;
        max-width: 150px;
        overflow: hidden;
        text-overflow: ellipsis;
        ",
            "@media (max-width: 767px)",
            " {
            font-size: 14px;
        }
    }
    .file-size {
        font-size: 16px;
        line-height: 36px;
        width: 83px;
        margin-left: 6px;
        white-space: nowrap;
    }
    .file-close {
        position: absolute;
        right: 13px;
        top: 50%;
        transform: translateY(-50%);
        cursor: pointer;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
